lib/9pfs/README.md

```diff
@@ -0,0 +1,167 @@
+  */
+ char                   *readdir_buf;
+ /*
+  * Offset within the buffer where the `uk_9p_stat`
+  * stat structure of the next child can be found
+  */
+ int                    readdir_off;
+ /* Total size of the data in the `readdir` buf */
+ int                    readdir_sz;
+};
+```
+
+The `uk_9pfs_mount_data` structure is used for operations that can be done on the files in the filesystem (e.g `open`, `close`, `readdir`).
+It contains:
+
+* The `fid`, a unique file identifier which identifies the file handle.
+* The `readdir_buf`, which contains the information read during the `readdir()` calls.
+  In the `uk_9pfs_mount_data` we can also find the `readdir_sz` which is the size of data in the `readdir_buf`.
+* The `readdir_off`, which refers to the offset from within the `readdir_buf` where we can find information about the next child.
+
+### Note Data Structure
+
+```c
+struct uk_9pfs_node_data {
+ /* File id of the vfs node */
+ struct uk_9pfid        *fid;
+ /* Number of files opened from the vfs node */
+ int                    nb_open_files;
+ /* Is a 9P remove call required when `nb_open_files` reaches 0? */
+ bool                   removed;
+};
+```
+
+The `uk_9pfs_node_data` structure is used for operations that involve allocating or freeing `vnode` data.
+
+## Configuring Applications to Use `9pfs`
+
+An Unikraft application that uses the `9pfs` filesystem will use code snippets such as the one below:
+
+```c
+#include <stdio.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+int main(int argc, char *argv[])
+{
+   char buf[1024];
+   int fd;
+
+   fd = open("/westworld.txt", O_RDWR | O_CREAT, 0777);
+   write(fd, "These violent delights have violent ends.", 41);
+   close(fd);
+
+   fd = open("/westworld.txt", O_RDWR | O_CREAT, 0777);
+   read(fd, buf, 41);
+   printf("%s\n", buf);
+   close(fd);
+
+   return 0;
+}
+```
+
+To configure the application to run under `9pfs` we follow the steps below:
+
+1. Enter the configuration interface by running:
+
+   ```console
+   $ make menuconfig
+   ```
+
+1. Under `Library Configuration`, select `vfscore:` `VFS Core Interface`.
+1. Under `vfscore: Configuration  --->` select `Automatically mount a root filesystem (/)`.
+1. Select `Default root filesystem -> 9pfs`.
+1. Set the `Default root device` option to `rootfs` (which is the default value).
+1. We want to run Unikraft with QEMU / KVM, so we must select KVM guest in the `Platform Configuration` menu.
+For `9pfs` we also need to enable, in the KVM guest options menu, `Virtio --->`, then `Virtio PCI device support`.
+1. Save the configuration and exit the interface.
+1. Create a new folder in the current directory by running:
+
+   ```console
+   $ mkdir rootfs
+   ```
+
+The complete command for running this program with `qemu` is:
+
+```console
+$ qemu-system-x86_64 -fsdev local,id=myid,path=./rootfs/,security_model=none -device virtio-9p-pci,fsdev=myid,mount_tag=rootfs -kernel build/06-adding-filesystems_kvm-x86_64 -nographic
```

Comment:
```suggestion
$ qemu-system-x86_64 -fsdev local,id=myid,path=./rootfs/,security_model=none -device virtio-9p-pci,fsdev=myid,mount_tag=rootfs -kernel build/app-helloworld_kvm-x86_64 -nographic
```
Change the image name to the one described below in the command options.

Reply:
You could also break this command in more lines using `\` if you think it looks better. If not, just leave it like this.